src/Hosting/TestHost/src/TestServer.cs

```diff
@@ -21,20 +21,31 @@ public class TestServer : IServer
         private IHttpApplication<Context> _application;
 
         /// <summary>
-        /// For use with IHostBuilder or IWebHostBuilder.
+        /// For use with IHostBuilder.
         /// </summary>
         public TestServer()
-            : this(new FeatureCollection())
+            : this(new ServiceCollection().BuildServiceProvider())
```

Comment:
Whats the point of this constructor?

Reply:
I'm not sure if it can be removed. The parameter-less constructor already existed before this PR.

Reply:
Where is the constructor with the IServiceProvider used?

Reply:
This was the one used by DI in the HostBuilder scenario. We could obsolete it with a warning if you like @davidfowl?

Reply:
The IServiceProvider ctor is the new one used by DI in the HostBuilder scenario.

Reply:
@huysentruitw don't wait for David, he's traveling. We can obsolete old things later if we need to.

Reply:
We should obsolete them if they aren't useful anymore (or don't work), otherwise it's fine keeping it.

Reply:
Looking back at 2.2, these weren't even there, they were added in 3.0 for HostBuilder. Ok, we can remove them.
https://github.com/aspnet/AspNetCore/blob/release/2.2/src/Hosting/TestHost/src/TestServer.cs

I'll comment on the individual APIs to remove.